#16 useUmConsent configurable rootDomain

jaydonkrooss opened the pull request:
add rootDomain, potentially preventing the bug of the value being true leading to an invalid domain

jaydonkrooss committed:
#16 useUmConsent configurable rootDomain

pushyamig commented:
I will get to this review this afternoon.

jonespm commented:
This looks good, might also want to add this to the example in the README.

I believe if you this was set to `true` here there would still be a problem, but maybe not worth checking for that.

jaydonkrooss committed:
Use null coalesce to avoid confusion, update readme for proper explanation

jaydonkrooss committed:
readme tweak

jaydonkrooss commented:
> This looks good, might also want to add this to the example in the README.
> 
> I believe if you this was set to `true` here there would still be a problem, but maybe not worth checking for that.

@jonespm I changed this so that the module does not accept a non-string value. also added a more in-depth explanation of the possible parameters in the readme.

pushyamig commented on src/useUmConsent.ts:
```diff
@@ -84,8 +86,8 @@ export function useUmConsent() : UmConsentHookReturn {
             customManager: {
                 enabled: true,
-                alwaysShow: alwaysShow || false,
-                rootDomain: true,
```
Do it make sense to write test cases supporting this?

jaydonkrooss replied:
created an issue to address:  https://github.com/tl-its-umich-edu/react-ga-onetrust-consent/issues/18

pushyamig commented:
Ok, I feel I can't test this change from `local-app` since the change before this and here works in same way, once cookies accepted the Banner won't pop-up. Once this is merged we can retest it to various project integrations of this package. 

I have one comment with Unit test supporting this use case since this can't have tested from UI. But other than that this could be merged

jonespm commented:
I tested this with the sample app without setting any value for this rootDomain and it set the cookie fine as expected now. One accepted it works fine. For some reason if I provide a value of "site" here it's not creating any cookies. Still looking into this. 

um_privacy_consent_ee0282041c3c...

jonespm reviewed (approved):
Tested to work fine. I think it would make testing easier if there was a way to pass config changes into the test app here that can be created as a separate issue. Maybe easiest with a json file in the public directory?

pushyamig commented:
I have my comment, i will defer to @jonespm approval on this and can be merged if my comments are not resolved

jonespm commented:
@pushyamig Passing this value is optional. If you don't specify it, it will default to the entire domain which works.

I think you'd only pass a value if you had multiple domains that you wanted this privacy banner to support. And it looks like it only goes one level.

Like we if wanted CCM and MyLA to be under the same banner and the user only has to agree to one of them and not both. However it feels like we may want to just have them agree to both. Then we'd never change this value from false.

pushyamig commented:
@jonespm Thanks for clarifying on this. I will keep this in mind with reviewing PR with this integration.

jaydonkrooss commented:
I created issues to improve the testing of these configurations (both local and unit tests). Good feedback, thanks!
